lib/plugins/aws/package/lib/saveCompiledTemplate.js

```diff
@@ -13,8 +13,20 @@ module.exports = {
       compiledTemplateFileName
     );
 
-    this.serverless.utils.writeFileSync(compiledTemplateFilePath,
-      this.serverless.service.provider.compiledCloudFormationTemplate);
+    const compiledTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+    if (compiledTemplate.Resources[this.provider.naming.getPolicyLogicalId()]) {
+      const customIAMPolicyStatement = compiledTemplate
+        .Resources[this.provider.naming.getPolicyLogicalId()]
+        .Properties
+        .PolicyDocument
+        .Statement;
+      // remove custom IAM Policy if no custom statements (empty array would be invalid for CF)
+      if (!customIAMPolicyStatement.length) {
+        delete compiledTemplate.Resources[this.provider.naming.getPolicyLogicalId()];
```

Comment:
Is this ok to introduce further changes to compiledTemplate here? Maintance-wise it can be confusing.
So far it was purely about storing compiled template into file, now there's also a cleanup action added here.

Maybe we should rather introduce some dedicated _cleanup/optimize step_, prior _save_ that one, and put it there to make it transparent?